Reformatting Factory and Cove directory and naming conventions

PolishUser opened the pull request:
- makes file structure and naming conventions consistent
- updates cove hero ordering
- adds a submod for factory campaign heroes to enable them for random maps

If you merge this, merge those too so that they don't get broken:
https://github.com/vcmi-mods/tarnum/pull/14
https://github.com/vcmi-mods/factory-creatures/pull/38
https://github.com/vcmi-mods/new-old-heroes/pull/11
https://github.com/vcmi-mods/pvp-balance/pull/6

Commit:
Cleanup Part 1

Commit:
Cleanup Part 1

Commit:
Cleanup Part 1

Commit:
Cleanup Part 1

Commit:
Cleanup Part 1

Commit:
Cleanup Part 1

Commit:
Cleanup Part 1

Commit:
Cleanup Part 1

Commit:
Cleanup Part 2

Commit:
Cleanup Part 2

Commit:
Cleanup Part 2

Commit:
Cleanup Part 2

Commit:
Cleanup Part 2

Commit:
Cleanup Part 2

Commit:
Cleanup Part 2

Commit:
Cleanup Part 2

Commit:
Fix Minor Typos

Commit:
Fix Minor Typos

Commit:
Fix Minor Typos

Commit:
Fix Minor Typos

Commit:
Fix Minor Typos

Commit:
Fix Minor Typos

Commit:
Fix Minor Typos

Commit:
Fix Typos and Structure

Commit:
Fix Typos and Structure

Commit:
Fix Typos and Structure

Commit:
Fix Typos and Structure

Commit:
Fix Typos and Structure

Commit:
Fix Typos and Structure

Commit:
Fix Typos and Structure

Commit:
Fix Typos and Structure

GeorgeK1ng commented:
Is your repo ready for testing?

Commit:
Fix Filepaths

Commit:
Fix Filepaths

Commit:
Fix Filepaths

Commit:
Fix Filepaths

PolishUser commented:
> Is your repo ready for testing?

I've just done final fixes to the filepaths in the config .jsons. I think that it should be ready now.

PolishUser committed:
Merge branch 'vcmi-1.6' into huge-cleanup

GeorgeK1ng commented:
Ok, I will check it in few hours.  Please check main repo with latest applied fixes aswell.

Commit:
Add Mod Version Info

DjWarmonger committed:
Make columns static so they can be used by RMG

DjWarmonger committed:
Merge remote-tracking branch 'origin/vcmi-1.6' into 1_70_biomes_fix

# Conflicts:
#	hota/mod.json

Commit:
Translations and ensuring consistent hero format

Commit:
Translations and ensuring consistent hero format

Commit:
Translations and ensuring consistent hero format

Commit:
Translations and ensuring consistent hero format

Commit:
cowboy translation fix

Commit:
cowboy translation fix

Commit:
cowboy translation fix

Commit:
cowboy translation fix

Commit:
Add new campaign heroes and improve heroes submod format

Commit:
fix submod desc

Commit:
Fix typos

Commit:
fix balfour typo

Commit:
Minor fixes

Commit:
Add ballista for gruezak

Commit:
Pluralise Specialities

Commit:
Fix specialty typo

Commit:
Fix specialty typo

Commit:
Fix specialty typo

Commit:
Fix specialty typo

Commit:
Fix specialty typo

Commit:
Fix specialty typo

Commit:
Fix specialty typo

Commit:
Fix specialty typo

Commit:
fix special flag disabler submod

Commit:
fix special flag disabler submod

Commit:
fix special flag disabler submod

Commit:
fix special flag disabler submod

Commit:
fix special flag disabler submod

Commit:
fix special flag disabler submod

Commit:
fix special flag disabler submod

Commit:
fix special flag disabler submod

Commit:
fix true flag

Commit:
fix true flag

Commit:
fix true flag

Commit:
fix true flag

Commit:
fix true flag

Commit:
fix true flag

Commit:
fix true flag

Commit:
fix true flag

GeorgeK1ng commented:
Why we need 2 "Campaing heroes" submods? Why separating them? One campaign hero unlocked sounds for me enough. It's still Horn of the Abyss Campaign. I don't see reason for adding another submod.

Your changes also breaks HotA map support. I'm no longer able to load any HotA map with your touches..

PolishUser commented:
> Why we need 2 "Campaing heroes" submods? Why separating them? One campaign hero unlocked sounds for me enough. It's still Horn of the Abyss Campaign. I don't see reason for adding another submod.

Thanks for your feedback!
I've decided to split them since HotA's factory campaign added a ton of 'portrait heroes' which are just copy and pasted factory heroes and I thought that we should give the option for the player to enable them in random maps/single scenarios.

> 
> Your changes also breaks HotA map support. I'm no longer able to load any HotA map with your touches..

Will take a look at HotA map support.

PolishUser commented:
> Your changes also breaks HotA map support. I'm no longer able to load any HotA map with your touches..

Weirdly enough HotA map support works for me just fine (I can open maps both in editor and game). Could you send me your logs and tell which maps have you been trying to test/open? I am testing with HotA and Vcmi-extras only.

Commit:
Fix artificer animation sprites

Commit:
Fix artificer animation sprites

Commit:
Fix artificer animation sprites

Commit:
Fix artificer animation sprites

GeorgeK1ng commented:
![image](https://github.com/user-attachments/assets/eaa3d4f6-35d4-4577-bada-8a088be311a8)

Try for example this map. This happends only when I swap hota mod folder with your ones.

GeorgeK1ng commented:
And this is my mods preset, whichhave no issues unless using your modded hota.

```
{
	"mods" : [ "vcmi", "core", "boost-ai", "extreme-ai", "combine-grail", "fused-artifacts", "moon-artifacts", "witchking-artifacts", "ab-bad-ending-assets", "andruids-expansion", "morns-battlefields", "neutral-heroes", "forge", "creatures-hidden-potential", "object-pack", "modder-tools-pack", "skills-pack", "adventure-spells-pack", "andruids-spell-balance", "new-old-spells-plus", "new-summon-spells", "abyss-town", "haven-town", "tavern-invite", "zefix", "vivid-battlefields", "lotrd-townscreens", "alternate-townscreens", "asphalt-terrain", "movement-arrows", "highlands-town", "chronicles", "courtyard", "death-valley-town", "small-era-mods", "tarnum", "vcmi-extras", "wake-of-gods", "new-pavilion", "newtown-terrains", "tides-of-war", "campaign-heroes", "new-monsters-pack", "czech-translation", "hota" ],
	"name" : "default",
	"settings" : {
		"ab-bad-ending-assets" : { "ab-bad-ending-maps" : false },
		"abyss-town" : {
			"changed banks" : true,
			"new battle music" : true
		},
		"adventure-spells-pack" : {
			"air-spells" : true,
			"earth-spells" : true,
			"fire-spells" : true,
			"global-spells" : true,
			"water-spells" : true
		},
		"alternate-townscreens" : {
			"castle" : false,
			"castle.red-castle" : false,
			"castle.snow-castle" : false,
			"ghost-necropolis" : false,
			"mirror-towns" : true,
			"mirror-towns.castle" : false,
			"mirror-towns.conflux" : false,
			"mirror-towns.dungeon" : false,
			"mirror-towns.fortress" : false,
			"mirror-towns.inferno" : false,
			"mirror-towns.necropolis" : false,
			"mirror-towns.rampart" : false,
			"mirror-towns.stronghold" : false,
			"mirror-towns.tower" : false,
			"rampart" : false,
			"rampart.another-rampart" : false,
			"rampart.autumn-rampart" : false,
			"tower" : true,
			"tower.sand-tower" : false,
			"tower.snow-in-tower" : true
		},
		"andruids-expansion" : {
			"damned utopia" : true,
			"initiative" : true,
			"minotaur warlock" : true,
			"minotaur warlock.new mino warlock" : true,
			"plaguetent" : true,
			"powerbond" : true
		},
		"andruids-spell-balance" : {
			"enhancedeagleeye" : true,
			"level2slow" : true,
			"massspells" : true,
			"massspells.magicshool" : true,
			"massspells.wisdom+magicschool" : false,
			"spellsrange" : true
		},
		"boost-ai" : {
			"ai speed up - limit on map heroes" : true,
			"battle bonuses for ai" : true,
			"global bonuses for ai" : true,
			"neutral_stack_exp" : true,
			"nkai experimental" : true,
			"nkai speed up - limit on map heroes" : true,
			"resourceful ai" : true
		},
		"chronicles" : {
			"chronicles_1" : true,
			"chronicles_2" : true
		},
		"creatures-hidden-potential" : {
			"alternative shooting centaur" : true,
			"enraged minotaur" : true,
			"furious cyclops" : true,
			"karmic harpies" : true,
			"siege dendroids" : true,
			"teleporting elemental" : true
		},
		"forge" : { "lessrandomstart" : true },
		"fused-artifacts" : {
			"fusedartifactshota" : true,
			"fusedartifactssod" : true
		},
		"hota" : {
			"artifacts" : true,
			"cannon" : true,
			"changedshootingvisuals" : true,
			"confluxvaultofashes" : true,
			"cove" : true,
			"covecampaignheroesinmaps" : true,
			"factory" : true,
			"factorycampaignheroesinmaps" : true,
			"gamebalance" : true,
			"heroes" : true,
			"heroes3datapatch" : true,
			"heroes3datapatch.arrow" : false,
			"heroesorchestracovetheme" : true,
			"highlandsterrain" : true,
			"interference" : true,
			"lessrandomstart" : true,
			"mainmenu" : false,
			"mainmenu.mainmenulegacy" : false,
			"mapdecorations" : true,
			"mapobjects" : true,
			"maps" : true,
			"mapsupport" : true,
			"neutralcreatures" : true,
			"newgraphics" : true,
			"spellresearch" : true,
			"supplementary" : true,
			"templates" : true,
			"terrainoverlays" : true,
			"wastelandterrain" : true
		},
		"lotrd-townscreens" : {
			"castle" : false,
			"conflux" : false,
			"dungeon" : true,
			"fortress" : false,
			"inferno" : false,
			"necropolis" : true,
			"rampart" : false,
			"stronghold" : false,
			"tower" : false
		},
		"movement-arrows" : {
			"hota_blue" : false,
			"hota_cyan" : false,
			"hota_magenta" : false,
			"hota_neon_cyan" : false,
			"hota_neon_green" : true,
			"hota_orange" : false,
			"hota_pink" : false,
			"hota_yellow" : false,
			"hota_yellow-orange" : false
		},
		"neutral-heroes" : {
			"hotaneutralheroes" : true,
			"wogneutralheroes" : true
		},
		"new-monsters-pack" : {
			"axolotl-creatures-pack" : true,
			"carpet-whisperers" : true,
			"doommod" : true,
			"dwarf-pack" : true,
			"extra-creature-pack" : true,
			"flugel" : true,
			"invisible-man" : true,
			"kurek-creatures" : true,
			"kurek-creatures.new-content-for-castle" : true,
			"kurek-creatures.new-content-for-castle.cherub" : true,
			"kurek-creatures.new-content-for-fortress" : true,
			"kurek-creatures.new-content-for-fortress.black-gorgon" : true,
			"kurek-creatures.new-content-for-haven" : true,
			"kurek-creatures.new-content-for-haven.marine-serail" : true,
			"kurek-creatures.new-content-for-neutral" : true,
			"kurek-creatures.new-content-for-neutral.eagle" : true,
			"kurek-creatures.new-content-for-neutral.orangedragon" : true,
			"kurek-creatures.new-content-for-neutral.purpledragon" : true,
			"kurek-creatures.new-content-for-neutral.wildgriffin" : true,
			"kurek-creatures.new-content-for-pavilion" : true,
			"kurek-creatures.new-content-for-pavilion.pavilionpharaoh" : true,
			"kurek-creatures.new-content-for-stronghold" : true,
			"kurek-creatures.new-content-for-stronghold.bloodeyed_cyclops" : true,
			"kurek-creatures.new-content-for-stronghold.cyclops_warrior" : true,
			"lord-hawk-creatures-pack" : true,
			"lord-hawk-creatures-pack.dark elemental" : true,
			"lord-hawk-creatures-pack.man at arms" : true,
			"lord-hawk-creatures-pack.mountain dwarf" : true,
			"tok" : true,
			"tok.more creatures" : true,
			"tok.more heroes" : false,
			"tok.more heroes.hota heroes" : false,
			"tok.more heroes.neutral classes" : false,
			"tok.more heroes.tok heroes" : false,
			"tok.more heroes.tow heroes" : false,
			"tok.more heroes.vanilla heroes" : false,
			"tok.more secondary" : true,
			"undeadsphinxes" : true,
			"wicked-witch" : true
		},
		"new-pavilion" : {
			"desert objects" : true,
			"dunes-terrain" : true
		},
		"new-summon-spells" : {
			"onlyonesummon" : false,
			"onlyonesummon.summonasylumcreatures" : false,
			"onlyonesummon.summonhotacreatures" : false,
			"onlyonesummon.summontowcreatures" : false,
			"summonasylumcreatures" : false,
			"summonhotacreatures" : true,
			"summontowcreatures" : true
		},
		"newtown-terrains" : {
			"deadlandterrain" : true,
			"deadlandterrain.death-valley-connector" : true,
			"fieldgloryterrain" : true,
			"fierylandterrain" : true,
			"heatherthicket" : true,
			"heatherthicket.grove-connect" : false,
			"icelandterrain" : true,
			"meadlandterrain" : true,
			"plateaulandterrain" : true,
			"plateaulandterrain.refugee-connect" : false,
			"shallowsterrain" : true,
			"shallowsterrain.abyss-connect" : true,
			"spoiledland" : true,
			"spoiledland.farriery-connect" : false,
			"stardustterrain" : true,
			"terraceland" : true,
			"terraceland.retreat-connect" : false
		},
		"object-pack" : {
			"abaddon-column" : true,
			"ballista-bank" : true,
			"central-warehouse" : true,
			"elemental-nodes" : true,
			"golem-factory" : false,
			"modern_roads-mod" : true
		},
		"skills-pack" : {
			"endurance" : false,
			"fame" : false,
			"four-schools" : true,
			"four-schools.disablemagic" : false,
			"magic-fader" : true,
			"warding" : false
		},
		"small-era-mods" : {
			"bubble arrows" : false,
			"darkened players" : false,
			"difficulty buttons" : true,
			"new flags" : false,
			"new grids" : true,
			"new grids.01" : true,
			"new grids.02" : false,
			"new grids.03" : false,
			"new grids.04" : false,
			"new grids.05" : false,
			"new terrains" : false,
			"no flashing trees" : false,
			"tavernvideos" : true,
			"unmounted heroes" : false
		},
		"tarnum" : {
			"tarnumcourtyard" : true,
			"tarnumcove" : true,
			"tarnumdriter" : true,
			"tarnumfactory" : true,
			"tarnumfactoryunofficial" : false,
			"tarnumfairy" : false,
			"tarnumforge" : true,
			"tarnumpavillion" : true,
			"tarnumrefugee" : false,
			"tarnumretreat" : false
		},
		"tides-of-war" : {
			"alternative-creatures" : false,
			"artifacts" : true,
			"banks" : false,
			"heroes" : false,
			"hota-balance-compatibility-patch" : false,
			"lessrandomstart" : true,
			"neutral-creatures" : true,
			"objects" : true,
			"skills" : true,
			"spells" : true,
			"tow-main-menu" : false,
			"transformationaltar" : false
		},
		"vcmi-extras" : {
			"adventuremap" : true,
			"arrowtowericons" : true,
			"battlefieldactions" : true,
			"bonusicons" : true,
			"bonusicons.bonus icons" : true,
			"bonusicons.immunity icons" : true,
			"chroniclesicon" : true,
			"extendedlobby" : true,
			"quick-exchange" : true
		},
		"vivid-battlefields" : { "lowlevelguards" : true },
		"wake-of-gods" : {
			"animatedobjects" : true,
			"animatedtrees" : true,
			"artifacts" : true,
			"battledecorations" : true,
			"commanders" : true,
			"creaturebanks" : true,
			"creatures" : true,
			"h2icons" : false,
			"heroes3datapatch" : true,
			"level8units" : true,
			"mainmenu" : true,
			"mapdecorations" : true,
			"mapobjects" : true,
			"mapsupport" : true,
			"mapsupport.commanderssupport" : true,
			"mapsupport.stackartifactssupport" : true,
			"mapsupport.warmachinesonmapsupport" : true,
			"messengersandemissaries" : true,
			"messengersandemissaries.vivid-battlefields" : true,
			"stackartifacts" : true,
			"stackexperience" : true,
			"stackexperience.stackexperienceicons" : false,
			"vividbattlefields" : true,
			"vividbattlefields.vivid-battlefields-8th-level" : true,
			"warmachinesonmap" : true,
			"woggraphicfix" : true,
			"woggraphicfix.wf_arrow" : false,
			"woggraphicfix.wf_artifacts" : true,
			"woggraphicfix.wf_commanders" : true,
			"woggraphicfix.wf_creatures" : true,
			"woggraphicfix.wf_h3creatures" : true,
			"woggraphicfix.wf_heroes" : true,
			"woggraphicfix.wf_level8units" : true,
			"woggraphicfix.wf_mapobjects" : true,
			"woggraphicfix.wf_messengersandemissaries" : true
		}
	}
}
```

PolishUser commented:
> ![image](https://private-user-images.githubusercontent.com/98261225/399814354-eaa3d4f6-35d4-4577-bada-8a088be311a8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU4NTQ5MTUsIm5iZiI6MTczNTg1NDYxNSwicGF0aCI6Ii85ODI2MTIyNS8zOTk4MTQzNTQtZWFhM2Q0ZjYtMzVkNC00NTc3LWJhZGEtOGEwODhiZTMxMWE4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTAyVDIxNTAxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYwMTVlYmVkY2I4N2NhMTc5NjkyMDE5Mjg0ZGI5ZjZiMTdmNGE0M2FhNmFjYzZmMWZmYTU5MDc4MTJhY2FkMjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Z76iaSXGzgyCCoi3WddNqGfRgf8CdP_EKhpPIXkSumo)
> 
> Try for example this map. This happends only when I swap hota mod folder with your ones.

![image](https://github.com/user-attachments/assets/a6515253-8834-45af-b28d-d9b7ae062acf)

Sure did, however, it works in my case. I can see that you are using a beta branch, may be a problem related to that? Also another question, did you just drag'n'drop my hota branch or deleted your older hota and replaced with mine? You could also try disabling the "Tarnum" mod due to the fact that I have changed factory hero classes and there may be a conflict since the 'factory' Tarnum variant may still be using older heroclass format (I am 90% certain that this mod is causing the crash).

GeorgeK1ng committed:
Update mod.json

Other -> Maps

Commit:
Update Factory Hero reorganisation + translations

Commit:
Update Factory Hero reorganisation + translations

Commit:
Update Factory Hero reorganisation + translations

Commit:
Update Factory Hero reorganisation + translations

Commit:
Update Factory Hero reorganisation + translations

Commit:
Update Factory Hero reorganisation + translations

Commit:
Update Factory Hero reorganisation + translations

Commit:
Update Factory Hero reorganisation + translations

GeorgeK1ng committed:
Artilery cannon fix

GeorgeK1ng committed:
Update mod.json

Commit:
Update Cove Hero Ordering, Translations and Map Format

Commit:
Update Cove Hero Ordering, Translations and Map Format

Commit:
Update Cove Hero Ordering, Translations and Map Format

Commit:
Update Cove Hero Ordering, Translations and Map Format

misiokles committed:
Merge pull request #183 from vcmi-mods/1_70_biomes_fix

Make columns static so they can be used by RMG

misiokles committed:
Merge pull request #185 from GeorgeK1ng/vcmi-1.6

Update mod.json

Commit:
Merge branch 'huge-cleanup' of https://github.com/PolishUser/horn-of-the-abyss into huge-cleanup

Commit:
Update mod.json

misiokles committed:
Merge pull request #184 from PolishUser/new-heroes

Add new campaign heroes and improve 'heroes submod' format

GeorgeK1ng commented:
Downloaded. Please change Interference mod type to "Skills" as minor fix.

Seems in Forge is same hero 13agar

```
Mod 'forge' have added object with name '13agar' that is also available in mod 'hota.factory'
Two objects with same name were loaded. Please use form 'hota.factory:13agar' if mod 'forge' needs to modify this object instead
Mod 'hota.factory' have added object with name '13agar' that is also available in mod 'forge'
Two objects with same name were loaded. Please use form 'forge:13agar' if mod 'hota.factory' needs to modify this object instead
```

Commit:
Merge branch 'huge-cleanup' of https://github.com/PolishUser/horn-of-the-abyss into huge-cleanup

Commit:
Update factory mod.json

Commit:
Update factory mod.json

Commit:
update mod.json

Commit:
update mod.json

Commit:
Merge branch 'huge-cleanup' of https://github.com/PolishUser/horn-of-the-abyss into huge-cleanup

Commit:
Minor Special Hero Unlock Fixes
